### PR TITLE
feat(endpoints): Inbound PUT authorizations handler 

### PR DESCRIPTION
**Pull request by sridharvoruganti**
`Ref story: #362`

- Handle inbound PUT /thirdpartyRequests/transactions/{id}/authorizations from the Auth Service
- handles errors using standard error callbacks
- Modified POST /thirdpartyRequests/transactions unit tests in accordance with authorizations

### CODE REVIEW

File: src/interface/api-template.yaml [hunk 1]
```diff
@@ -298,3 +352,36 @@ components:
...
+            This tells DFSP and auth-service which constent allows the PISP to initiate transaction.
+        sourceAccountId:
+          allOf:
+            - $ref: '#/components/schemas/AccountId'
+          description: DFSP specific account identifiers, e.g. `dfspa.alice.1234`
+        status:
+          type: string
+          enum:
+            - VERIFIED
+          description: The status of the authorization. This MUST be VERIFIED for a PUT request
```
Comment (lewisdaly) [inline src/interface/api-template.yaml:381]: So I think the logic here is that if there was a verification error, then the `PUT /thirdpartyRequests/transactions/{id}/authorizations/error` callback would be used instead? I think we should just find a similar request in the Mojaloop API spec to double check this pattern.

This is one of the drawbacks with our approach of worrying about the error conditions later on.

File: test/unit/index.test.ts [hunk 2]
```diff
@@ -227,10 +230,125 @@ describe('index', (): void => {
...
+        // Act
+        const response = await server.inject(request)
+
+        // Assert
+        expect(response.statusCode).toBe(202)
+        expect(response.result).toBeNull()
+        expect(mockForwardAuthorizationRequest).toHaveBeenCalledWith(...expected)
+      })
+
+      it('responds with a 400 when status !== VERIFIED', async (): Promise<void> => {
```
Comment (lewisdaly) [inline test/unit/index.test.ts:285]: See above comments on the `api-template.yaml`. I think this is right, but we need to think forward a little bit about error handling.

File: src/domain/thirdpartyRequests/authorizations.ts [hunk 3]
```diff
@@ -29,78 +30,74 @@ import Logger from '@mojaloop/central-services-logger'
...
  * @throws {FSPIOPError} Will throw an error if no endpoint to forward the transactions requests is
  *  found, if there are network errors or if there is a bad response
  * @returns {Promise<void>}
  */
-export async function forwardPostAuthorization (
+export async function forwardAuthorizationRequest (
   path: string,
+  endpointType: FspEndpointTypesEnum,
   headers: HapiUtil.Dictionary<string>,
+  method: string,
```
[the remark this answers is not in the document]
Comment (sridharvoruganti) [reply src/domain/thirdpartyRequests/authorizations.ts]: Ok, declared RestMethodsEnum in ambient.d.ts

File: test/unit/domain/thirdpartyRequests/authorizations.test.ts [hunk 4]
```diff
@@ -140,38 +145,168 @@ describe('domain/authorizations', () => {
...
+        Enum.Http.ResponseTypes.JSON,
+        undefined
+      ]
+      const sendRequestExpectedSecond = [
+        'http://pispA.local/thirdpartyRequests/transactions/123456/authorizations/error',
+        { 'fspiop-source': 'switch', 'fspiop-destination': 'pispA' },
+        'switch',
+        'pispA',
+        Enum.Http.RestMethods.PUT,
+        expect.any(Object),
```
[the remark this answers is not in the document]
Comment (sridharvoruganti) [reply test/unit/domain/thirdpartyRequests/authorizations.test.ts]: Fixed now, checking error object format

File: src/interface/api-template.yaml (hunk 1 above)
Comment (sridharvoruganti) [reply src/interface/api-template.yaml]: Ok

File: test/unit/index.test.ts (hunk 2 above)
Comment (sridharvoruganti) [reply test/unit/index.test.ts:285]: Ok

File: ambient.d.ts [hunk 5]
```diff
@@ -283,8 +285,20 @@ declare module '@mojaloop/central-services-shared' {
 }
 
 declare module '@mojaloop/central-services-error-handling' {
+  interface APIErrorObject {
+    errorInformation: {
+      errorCode?: string;
+      errorDescription?: string;
+      extensionList?: {
+        extension: [{
```
Comment (lewisdaly) [inline ambient.d.ts:293]: Is this the same as `Array<{key:string, value:string}>`?

**Review by lewisdaly — APPROVED** [review]

File: ambient.d.ts (hunk 5 above)
Comment (sridharvoruganti) [reply ambient.d.ts:293]: Yes

**Review by lewisdaly — APPROVED** [review]

**Pull request merged**